pyro/infer/abstract_infer.py

```diff
@@ -281,16 +281,18 @@ class TracePredictive(TracePosterior):
     achieved by constraining latent sites to randomly sampled parameter
     values from the model execution traces and running the model forward
     to generate traces with new response ("_RETURN") sites.
-
     :param model: arbitrary Python callable containing Pyro primitives.
-    :param TracePosterior posterior: trace posterior instance holding
-        samples from the model's approximate posterior.
+    :param TracePosterior posterior: trace posterior instance holding samples from the model's approximate posterior.
     :param int num_samples: number of samples to generate.
+    :param keep_sites: The sites which should be sampled from posterior predictive distribution (default: all)
+    :param keep_plates: The plates which should be sampled from posterior predictive distribution (default: all)
```

Comment:
@ahmadsalim - Thanks for the PR! It will be great if you could walk me through an example (maybe a toy model), and help me understand how this change helps the particular use case you have. While this looks mostly okay, I think I might be missing the larger picture of how you are using the `TracePredictive` class. We can sync offline too, if that helps.

Reply:
@neerajprad OK, thanks for the response 😄 . I will see if I can make a good minimal example.